src/main/java/io/anserini/index/generator/GeoGenerator.java

```diff
@@ -0,0 +1,85 @@
+package io.anserini.index.generator;
+
+import io.anserini.collection.JsonCollection;
+import io.anserini.index.IndexArgs;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.document.*;
+import org.apache.lucene.geo.Line;
+import org.apache.lucene.geo.Polygon;
+import org.apache.lucene.geo.SimpleWKTShapeParser;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+public class GeoGenerator implements LuceneDocumentGenerator<JsonCollection.Document> {
```

Comment:
Is this better named `WktGenerator`? Or maybe `WktGeometryGenerator`? Thoughts?

Reply:
Actually, it used to be named `WktGenerator`. But in order to make the raw data fit with the JsonCollection, the format is no longer `wkt`. It is more like a custom `json` file instead.